### ISSUE TITLE
Store paths are interpreted as flakes

### ISSUE THREAD

**Issue by danieldk** [NixOS/nix#4131]
**Describe the bug**

The `copy`, `path-info` and `copy` subcommands interpret store paths that contain `flake.nix` as flakes.

**Steps To Reproduce**

1. Create a store path that contains `flake.nix`.
2. Execute a command that we want to operate on a store path, e.g.:

```
$ nix sign-paths --key-file cache-priv-key.pem /nix/store/path-with-flake
$ nix path-info /nix/store/path-with-flake
```

3. Observe that the path is interpreted as a flake:

```
flake 'path:/nix/store/7zj2cs6ka320hplg4ia7rlmgy54z5war-source' does not provide attribute 'packages.x86_64-linux.defaultPackage.x86_64-linux', 'legacyPackages.x86_64-linux.defaultPackage.x86_64-linux' or 'defaultPackage.x86_64-linux'
```

**Expected behavior**

Treat the path as any other store path.

**`nix-env --version` output**

```
$ ~/git/nix/result/bin/nix-env --version
nix-env (Nix) 3.0pre19700101_58dadf2
```

**Comment by danieldk** [comment]
Still broken:

```
❯ nix path-info /nix/store/rhp35zbpdl7kzxn1ayjfp596qajgdjmi-source
error: flake 'path:/nix/store/rhp35zbpdl7kzxn1ayjfp596qajgdjmi-source' does not provide attribute 'packages.x86_64-linux.defaultPackage.x86_64-linux', 'legacyPackages.x86_64-linux.defaultPackage.x86_64-linux' or 'defaultPackage.x86_64-linux'
```

**Comment by edolstra** [comment]
Fixed by b2e8120d25de5303bfb6a99c14d3ecad21f55a05.
